services/dynamic/dynamic-regex.service.js

```
@@ -0,0 +1,99 @@
+        parameters: queryParams(
+          {
+            name: 'url',
+            description:
+              'The URL to a file to search. The full raw content will be used as the search string.',
+            required: true,
+            example:
+              'https://raw.githubusercontent.com/badges/shields/refs/heads/master/README.md',
+          },
+          {
+            name: 'search',
+            description:
+              'A re2 expression that will be used to extract data from the document. Only the first matched text will be returned.',
+            required: true,
+            example: 'Every month it serves (.*?) images',
+          },
+          {
+            name: 'replace',
+            description:
+              'A regex expression that will be used as the replacement of the search regex, like `$1` to specify the first matched group, etc. If empty (default), no replacement will be done and the full matched text will be shown.',
+            required: false,
+            example: '$1',
+          },
+          {
+            name: 'flags',
+            description:
+              'Flags to be used when creating the regex, like `i` for case insensitive, or `m` for multiline. None by default.',
+            required: false,
+            example: 'imsU',
+          },
+          {
+            name: 'noMatch',
+            description:
+              'String to be returned if the regex does not match the input. Empty by default.',
+            required: false,
+          },
+        ),
+      },
+    },
+  }
+  static defaultBadgeData = { label: 'match' }
+
+  async handle(namedParams, { url, search, replace, flags, noMatch = '' }) {
+    // fetch file
+    const { buffer } = await this._request({ url })
+
+    // build re2 regex
+    let re2
+    try {
+      re2 = new RE2(search, flags)
+    } catch (e) {
```

This will catch any exception.
Can you modify this to only do this if the original error is `SyntaxError`
If we get anything else, lets let it throw unhandled to start with and if anything pops up in Sentry after we deploy it we'll work out where next..

---

Also, can you add a test case for the "invalid regex" case